app/src/org/commcare/android/adapters/HomeScreenAdapter.java

```diff
@@ -0,0 +1,186 @@
+package org.commcare.android.adapters;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+
+import org.commcare.android.view.SquareButtonWithNotification;
+import org.commcare.dalvik.R;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+
+/**
+ * Created by dancluna on 3/19/15.
+ */
+public class HomeScreenAdapter extends BaseAdapter {
+    //region Buttons
+
+    static final int[] buttonsResources = new int[]{
+            R.layout.home_start_button,
+            R.layout.home_savedforms_button,
+            R.layout.home_incompleteforms_button,
+            R.layout.home_sync_button,
+            R.layout.home_disconnect_button,
+    };
+
+    static final HashMap<Integer, Integer> buttonsIDsToResources = new HashMap<Integer, Integer>() {{
+                put(R.id.home_start_sqbn,R.layout.home_start_button);
+                put(R.id.home_savedforms_sqbn,R.layout.home_savedforms_button);
+                put(R.id.home_sync_sqbn,R.layout.home_sync_button);
+                put(R.id.home_disconnect_sqbn,R.layout.home_disconnect_button);
+                put(R.id.home_incompleteforms_sqbn,R.layout.home_incompleteforms_button);
+    }};
+
+    //endregion
+
+    //region Private variables
+
+    final View.OnClickListener[] buttonListeners = new View.OnClickListener[buttonsResources.length];
+
+    final SquareButtonWithNotification[] buttons = new SquareButtonWithNotification[buttonsResources.length];
+
+    private Context context;
+
+    private boolean[] hiddenButtons = new boolean[buttonsResources.length];
+    private boolean isInitialized = false;
+
+    private LinkedList<SquareButtonWithNotification> visibleButtons;
+
+    //endregion
+
+    //region Constructors
+
+    public HomeScreenAdapter(Context c) { this.context = c; }
+
+    //endregion
+
+    //region Public API
+
+    public void setOnClickListenerForButton(int androidCode, boolean lookupID, View.OnClickListener listener){
+        int buttonIndex = getButtonIndex(androidCode, lookupID);
+        setOnClickListenerForButton(buttonIndex, listener);
+    }
+
+    public SquareButtonWithNotification getButton(int androidCode, boolean lookupID){
+        return buttons[getButtonIndex(androidCode, lookupID)];
+    }
+
+    public void setNotificationTextForButton(int androidCode, boolean lookupID, String notificationText) {
+        SquareButtonWithNotification button = getButton(androidCode, lookupID);
+        if (button != null) {
+            button.setNotificationText(notificationText);
+            notifyDataSetChanged();
+        }
+    }
+
+    @Override
+    public int getCount() {
+//        return buttonsResources.length;
+        return visibleButtons == null ? buttonsResources.length : visibleButtons.size();
```

Comment:
Why isn't this just visibleBottons.size()? A comment noting the phase shift from using buttonResources.length to visibleButtons.size() would be helpful.